[BUG]: "Wrong number of template arguments" compilation failure with thrust::copy and cuda::std::complex<double>

### Is this a duplicate?

- [x] I confirmed there appear to be no [duplicate issues](https://github.com/NVIDIA/cccl/issues) for this bug and that I agree to the [Code of Conduct](CODE_OF_CONDUCT.md)

### Type of Bug

Compile-time Error

### Component

Thrust

### Describe the bug

Hitting this compilation error with CCCL 3.1:

```
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include/cuda/std/__type_traits/type_list.h:82:9: error: wrong number of template arguments (1, should be at least 2)
   82 |   using __call _CCCL_NODEBUG_ALIAS = _Fn<_Ts...>;
      |         ^~~~~~
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include/cuda/std/__type_traits/type_list.h:517:1: note: provided for 'template<class, class _Ty, class ...> using __call = _Ty'
  517 | _CCCL_PP_REPEAT_REVERSE(_CCCL_META_UNROLL_LIMIT, _M1)
      | ^~~~~~~~~~~~~~~~~~~~~~~
```

### How to Reproduce

Reproducer:

```
#include <vector>
#include <cuda/std/complex>
#include <thrust/system/omp/execution_policy.h>
int main()
{
  using VAL = cuda::std::complex<double>;
  std::vector<VAL> ins;
  ins.push_back(VAL(1, 1));
  std::vector<VAL> outs;
  outs.push_back(VAL(0, 0));
  thrust::copy(thrust::omp::par, ins.data(), ins.data() + 1, outs.data());
}
```

Compilation command:

```
/usr/bin/c++ -DTHRUST_DEVICE_SYSTEM=THRUST_DEVICE_SYSTEM_CUDA -DTHRUST_HOST_SYSTEM=THRUST_HOST_SYSTEM_CPP -I/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include -I/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/thrust -I/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/cub -isystem /usr/local/cuda/targets/x86_64-linux/include -ftemplate-backtrace-limit=0 -fopenmp 1103.cc
```

<details>
<summary>Compilation output</summary>

```
/usr/bin/c++ -DTHRUST_DEVICE_SYSTEM=THRUST_DEVICE_SYSTEM_CUDA -DTHRUST_HOST_SYSTEM=THRUST_HOST_SYSTEM_CPP -I/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include -I/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/thrust -I/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/cub -isystem /usr/local/cuda/targets/x86_64-linux/include -ftemplate-backtrace-limit=0 -fopenmp 1103.cc
In file included from /home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include/cuda/std/__type_traits/make_signed.h:29,
                 from /home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include/cuda/std/__iterator/incrementable_traits.h:33,
                 from /home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include/cuda/std/__iterator/concepts.h:40,
                 from /home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include/cuda/std/__fwd/subrange.h:23,
                 from /home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include/cuda/std/__fwd/get.h:27,
                 from /home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include/cuda/std/detail/libcxx/include/complex:246,
                 from /home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include/cuda/std/complex:25,
                 from 1103.cc:3:
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include/cuda/std/__type_traits/type_list.h: In substitution of ‘template<long unsigned int _DependentValue> template<template<template<class ...> class _Fn, class ... _Ts> template<long unsigned int _DependentValue> template<class ...> class _Fn, class ... _Ts> using __call = _Fn<_Ts ...> [with _Fn = cuda::std::__4::__detail::__type_index_small_size_fn<1>::__call; _Ts = {double}; long unsigned int _DependentValue = 8]’:
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include/cuda/std/__type_traits/type_list.h:89:7:   required by substitution of ‘template<class _Fn, class ... _Ts> using __type_call_indirect = typename cuda::std::__4::__detail::__type_call_indirect_fn<sizeof (cuda::std::__4::__type_list<_Fn, _Ts ...>*)>::__call<_Fn::template __call, _Ts ...> [with _Fn = cuda::std::__4::__detail::__type_index_small_size_fn<1>; _Ts = {double}]’
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include/cuda/std/__type_traits/type_list.h:530:9:   required by substitution of ‘template<class _Ip, class ... _Ts> using __call = cuda::std::__4::__type_call_indirect<cuda::std::__4::__detail::__type_index_small_size_fn<_Ip::value>, _Ts ...> [with _Ip = cuda::std::__4::integral_constant<long unsigned int, 1>; _Ts = {double}]’
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include/cuda/std/__type_traits/type_list.h:66:7:   required by substitution of ‘template<class _Fn, class ... _Ts> using __type_call = typename _Fn::__call<_Ts ...> [with _Fn = cuda::std::__4::__detail::__type_index_select_fn<true>; _Ts = {cuda::std::__4::integral_constant<long unsigned int, 1>, double}]’
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include/cuda/std/__type_traits/type_list.h:537:7:   required by substitution of ‘template<class _Ip, class ... _Ts> using __type_index = cuda::std::__4::__type_call<cuda::std::__4::__detail::__type_index_select_fn<(_Ip::value < 16)>, _Ip, _Ts ...> [with _Ip = cuda::std::__4::integral_constant<long unsigned int, 1>; _Ts = {double}]’
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include/cuda/std/__type_traits/type_list.h:548:7:   required by substitution of ‘template<long unsigned int _Ip, class ... _Ts> using __type_index_c = cuda::std::__4::__type_index<cuda::std::__4::integral_constant<long unsigned int, _Ip>, _Ts ...> [with long unsigned int _Ip = 1; _Ts = {double}]’
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include/cuda/std/__type_traits/type_list.h:558:9:   required by substitution of ‘template<long unsigned int _Ip> template<class ... _Ts> using __call = cuda::std::__4::__type_index_c<_Ip, _Ts ...> [with _Ts = {double}; long unsigned int _Ip = 1]’
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include/cuda/std/__type_traits/type_list.h:66:7:   required by substitution of ‘template<class _Fn, class ... _Ts> using __type_call = typename _Fn::__call<_Ts ...> [with _Fn = cuda::std::__4::__detail::__type_at_fn<1>; _Ts = {double}]’
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include/cuda/std/__type_traits/type_list.h:311:9:   required by substitution of ‘template<class ... _Ts> template<class _Fn, class ... _Us> using __call = cuda::std::__4::__type_call<_Fn, _Ts ..., _Us ...> [with _Fn = cuda::std::__4::__detail::__type_at_fn<1>; _Us = {}; _Ts = {double}]’
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include/cuda/std/__tuple_dir/make_tuple_types.h:52:9:   required from ‘struct cuda::std::__4::__make_tuple_types_flat<cuda::std::__4::complex<double>, cuda::std::__4::__tuple_indices<0, 1> >’
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include/cuda/std/__tuple_dir/make_tuple_types.h:74:9:   required from ‘struct cuda::std::__4::__make_tuple_types<cuda::std::__4::complex<double>&, 2, 0, true>’
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include/cuda/std/__tuple_dir/sfinae_helpers.h:100:8:   required from ‘struct cuda::std::__4::__tuple_constructible<cuda::std::__4::complex<double>&, cuda::std::__4::tuple<cuda::std::__4::complex<double>&, cuda::std::__4::complex<double>&>, true, true>’
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include/cuda/std/detail/libcxx/include/tuple:592:113:   required by substitution of ‘template<class _Alloc, class _Tuple, typename cuda::std::__4::enable_if<cuda::std::__4::__tuple_constructible<_Tuple, cuda::std::__4::tuple<cuda::std::__4::complex<double>&, cuda::std::__4::complex<double>&>, cuda::std::__4::__tuple_like_ext<typename cuda::std::__4::remove_reference<_Up>::type>::value, true>::value, int>::type <anonymous> > cuda::std::__4::__tuple_impl<cuda::std::__4::__tuple_indices<0, 1>, cuda::std::__4::complex<double>&, cuda::std::__4::complex<double>&>::__tuple_impl(cuda::std::__4::allocator_arg_t, const _Alloc&, _Tuple&&) [with _Alloc = cuda::std::__4::complex<double>; _Tuple = cuda::std::__4::complex<double>&; typename cuda::std::__4::enable_if<cuda::std::__4::__tuple_constructible<_Tuple, cuda::std::__4::tuple<cuda::std::__4::complex<double>&, cuda::std::__4::complex<double>&>, cuda::std::__4::__tuple_like_ext<typename cuda::std::__4::remove_reference<_Up>::type>::value, true>::value, int>::type <anonymous> = <missing>]’
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include/cuda/std/detail/libcxx/include/tuple:807:9:   required from ‘constexpr cuda::std::__4::tuple< <template-parameter-1-1> >::tuple(const _Tp& ...) [with _Constraints = cuda::std::__4::__tuple_constraints<cuda::std::__4::complex<double>&, cuda::std::__4::complex<double>&>; typename cuda::std::__4::enable_if<_Constraints::__implicit_variadic_copy_constructible, int>::type <anonymous> = 0; _Tp = {cuda::std::__4::complex<double>&, cuda::std::__4::complex<double>&}]’
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/thrust/thrust/iterator/detail/tuple_of_iterator_references.h:68:18:   required from ‘typename thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::detail::make_zip_iterator_base<IteratorTuple>::type::reference thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::zip_iterator<IteratorTuple>::dereference_impl(cuda::std::__4::index_sequence<Indices ...>) const [with long unsigned int ...Is = {0, 1}; IteratorTuple = cuda::std::__4::tuple<cuda::std::__4::complex<double>*, cuda::std::__4::complex<double>*>; typename thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::detail::make_zip_iterator_base<IteratorTuple>::type::reference = thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::detail::tuple_of_iterator_references<cuda::std::__4::complex<double>&, cuda::std::__4::complex<double>&>; super_t = thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::iterator_facade<thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::zip_iterator<cuda::std::__4::tuple<cuda::std::__4::complex<double>*, cuda::std::__4::complex<double>*> >, cuda::std::__4::tuple<cuda::std::__4::complex<double>, cuda::std::__4::complex<double> >, thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::system::cpp::detail::tag, thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::random_access_traversal_tag, thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::detail::tuple_of_iterator_references<cuda::std::__4::complex<double>&, cuda::std::__4::complex<double>&>, long int>; cuda::std::__4::index_sequence<Indices ...> = cuda::std::__4::integer_sequence<long unsigned int, 0, 1>]’
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/thrust/thrust/iterator/zip_iterator.h:234:28:   required from ‘typename thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::detail::make_zip_iterator_base<IteratorTuple>::type::reference thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::zip_iterator<IteratorTuple>::dereference() const [with IteratorTuple = cuda::std::__4::tuple<cuda::std::__4::complex<double>*, cuda::std::__4::complex<double>*>; typename thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::detail::make_zip_iterator_base<IteratorTuple>::type::reference = thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::detail::tuple_of_iterator_references<cuda::std::__4::complex<double>&, cuda::std::__4::complex<double>&>; super_t = thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::iterator_facade<thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::zip_iterator<cuda::std::__4::tuple<cuda::std::__4::complex<double>*, cuda::std::__4::complex<double>*> >, cuda::std::__4::tuple<cuda::std::__4::complex<double>, cuda::std::__4::complex<double> >, thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::system::cpp::detail::tag, thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::random_access_traversal_tag, thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::detail::tuple_of_iterator_references<cuda::std::__4::complex<double>&, cuda::std::__4::complex<double>&>, long int>]’
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/thrust/thrust/iterator/iterator_facade.h:204:25:   required from ‘static typename Facade::reference thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::iterator_core_access::dereference(const Facade&) [with Facade = thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::zip_iterator<cuda::std::__4::tuple<cuda::std::__4::complex<double>*, cuda::std::__4::complex<double>*> >; typename Facade::reference = thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::detail::tuple_of_iterator_references<cuda::std::__4::complex<double>&, cuda::std::__4::complex<double>&>]’
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/thrust/thrust/iterator/iterator_facade.h:374:45:   required from ‘thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::iterator_facade<Derived, Value, System, Traversal, Reference, Difference>::reference thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::iterator_facade<Derived, Value, System, Traversal, Reference, Difference>::operator*() const [with Derived = thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::zip_iterator<cuda::std::__4::tuple<cuda::std::__4::complex<double>*, cuda::std::__4::complex<double>*> >; Value = cuda::std::__4::tuple<cuda::std::__4::complex<double>, cuda::std::__4::complex<double> >; System = thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::system::cpp::detail::tag; Traversal = thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::random_access_traversal_tag; Reference = thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::detail::tuple_of_iterator_references<cuda::std::__4::complex<double>&, cuda::std::__4::complex<double>&>; Difference = long int; reference = thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::detail::tuple_of_iterator_references<cuda::std::__4::complex<double>&, cuda::std::__4::complex<double>&>]’
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/thrust/thrust/system/omp/detail/for_each.inl:71:15:   required from ‘RandomAccessIterator thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::system::omp::detail::for_each_n(execution_policy<System1>&, RandomAccessIterator, Size, UnaryFunction) [with DerivedPolicy = par_t; RandomAccessIterator = thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::zip_iterator<cuda::std::__4::tuple<cuda::std::__4::complex<double>*, cuda::std::__4::complex<double>*> >; Size = long int; UnaryFunction = thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::detail::unary_transform_functor<cuda::std::__4::__identity>]’
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/thrust/thrust/system/omp/detail/for_each.inl:81:33:   required from ‘RandomAccessIterator thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::system::omp::detail::for_each(execution_policy<System1>&, RandomAccessIterator, RandomAccessIterator, UnaryFunction) [with DerivedPolicy = par_t; RandomAccessIterator = thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::zip_iterator<cuda::std::__4::tuple<cuda::std::__4::complex<double>*, cuda::std::__4::complex<double>*> >; UnaryFunction = thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::detail::unary_transform_functor<cuda::std::__4::__identity>]’
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/thrust/thrust/detail/for_each.inl:46:18:   required from ‘InputIterator thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::for_each(const detail::execution_policy_base<DerivedPolicy>&, InputIterator, InputIterator, UnaryFunction) [with DerivedPolicy = system::omp::detail::par_t; InputIterator = zip_iterator<cuda::std::__4::tuple<cuda::std::__4::complex<double>*, cuda::std::__4::complex<double>*> >; UnaryFunction = detail::unary_transform_functor<cuda::std::__4::__identity>]’
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/thrust/thrust/system/detail/generic/transform.inl:58:47:   required from ‘OutputIterator thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::system::detail::generic::transform(thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::execution_policy<Derived>&, InputIterator, InputIterator, OutputIterator, UnaryFunction) [with DerivedPolicy = thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::system::omp::detail::par_t; InputIterator = cuda::std::__4::complex<double>*; OutputIterator = cuda::std::__4::complex<double>*; UnaryFunction = cuda::std::__4::__identity]’
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/thrust/thrust/detail/transform.inl:47:19:   required from ‘OutputIterator thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::transform(const detail::execution_policy_base<DerivedPolicy>&, InputIterator, InputIterator, OutputIterator, UnaryFunction) [with DerivedPolicy = system::omp::detail::par_t; InputIterator = cuda::std::__4::complex<double>*; OutputIterator = cuda::std::__4::complex<double>*; UnaryFunction = cuda::std::__4::__identity]’
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/thrust/thrust/system/detail/generic/copy.inl:49:27:   required from ‘OutputIterator thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::system::detail::generic::copy(thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::execution_policy<Derived>&, InputIterator, InputIterator, OutputIterator) [with DerivedPolicy = thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::system::omp::detail::par_t; InputIterator = cuda::std::__4::complex<double>*; OutputIterator = cuda::std::__4::complex<double>*]’
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/thrust/thrust/system/omp/detail/copy.inl:50:41:   required from ‘OutputIterator thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::system::omp::detail::copy(execution_policy<System1>&, InputIterator, InputIterator, OutputIterator) [with DerivedPolicy = par_t; InputIterator = cuda::std::__4::complex<double>*; OutputIterator = cuda::std::__4::complex<double>*]’
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/thrust/thrust/detail/copy.inl:44:14:   required from ‘OutputIterator thrust::THRUST_300002_SM___CUDA_ARCH_LIST___NS::copy(const detail::execution_policy_base<DerivedPolicy>&, InputIterator, InputIterator, OutputIterator) [with System = system::omp::detail::par_t; InputIterator = cuda::std::__4::complex<double>*; OutputIterator = cuda::std::__4::complex<double>*]’
1103.cc:14:15:   required from here
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include/cuda/std/__type_traits/type_list.h:82:9: error: wrong number of template arguments (1, should be at least 2)
   82 |   using __call _CCCL_NODEBUG_ALIAS = _Fn<_Ts...>;
      |         ^~~~~~
/home/mpapadakis/b/legate/arch-legate/cmake_build/_deps/cccl-src/libcudacxx/include/cuda/std/__type_traits/type_list.h:517:1: note: provided for ‘template<class, class _Ty, class ...> using __call = _Ty’
  517 | _CCCL_PP_REPEAT_REVERSE(_CCCL_META_UNROLL_LIMIT, _M1)
      | ^~~~~~~~~~~~~~~~~~~~~~~
```

</details>



### Expected behavior

Compilation should succeed

### Reproduction link

https://godbolt.org/z/f6a4f37Yq

### Operating System

Linux iblis 6.8.0-83-generic #83~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Tue Sep  9 18:19:47 UTC 2 x86_64 x86_64 x86_64 GNU/Linux

### nvidia-smi output

N/A

### NVCC version

`nvcc` is not used. This is a host compiler issue:

```
/usr/bin/c++ --version
c++ (Ubuntu 12.3.0-1ubuntu1~22.04.2) 12.3.0
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

--- a/libcudacxx/include/cuda/std/__tuple_dir/make_tuple_types.h
+++ b/libcudacxx/include/cuda/std/__tuple_dir/make_tuple_types.h
@@ -20,6 +20,7 @@
 #  pragma system_header
 #endif // no system header
 
+#include <cuda/__fwd/complex.h>
 #include <cuda/std/__fwd/array.h>
 #include <cuda/std/__fwd/complex.h>
 #include <cuda/std/__fwd/tuple.h>
@@ -68,6 +69,17 @@ struct __make_tuple_types_flat<array<_Vt, _Np>, __tuple_indices<_Idx...>>
 template <class _Vt, size_t... _Idx>
 struct __make_tuple_types_flat<complex<_Vt>, __tuple_indices<_Idx...>>
 {
+  static_assert(sizeof...(_Idx) == 2, "__make_tuple_types: complex has only 2 members");
+  template <size_t>
+  using __value_type = _Vt;
+  template <class _Tp, class _ApplyFn = __apply_cvref_fn<_Tp>>
+  using __apply_quals _CCCL_NODEBUG_ALIAS = __tuple_types<__type_call<_ApplyFn, __value_type<_Idx>>...>;
+};
+
+template <class _Vt, size_t... _Idx>
+struct __make_tuple_types_flat<::cuda::complex<_Vt>, __tuple_indices<_Idx...>>
+{
+  static_assert(sizeof...(_Idx) == 2, "__make_tuple_types: complex has only 2 members");
   template <size_t>
   using __value_type = _Vt;
   template <class _Tp, class _ApplyFn = __apply_cvref_fn<_Tp>>

--- a/libcudacxx/include/cuda/std/__tuple_dir/make_tuple_types.h
+++ b/libcudacxx/include/cuda/std/__tuple_dir/make_tuple_types.h
@@ -65,11 +65,13 @@ struct __make_tuple_types_flat<array<_Vt, _Np>, __tuple_indices<_Idx...>>
   using __apply_quals _CCCL_NODEBUG_ALIAS = __tuple_types<__type_call<_ApplyFn, __value_type<_Idx>>...>;
 };
 
-template <class _Vt>
-struct __make_tuple_types_flat<complex<_Vt>, __tuple_indices<0, 1>>
+template <class _Vt, size_t... _Idx>
+struct __make_tuple_types_flat<complex<_Vt>, __tuple_indices<_Idx...>>
 {
+  template <size_t>
+  using __value_type = _Vt;
   template <class _Tp, class _ApplyFn = __apply_cvref_fn<_Tp>>
-  using __apply_quals _CCCL_NODEBUG_ALIAS = __tuple_types<__type_call<_ApplyFn, _Vt>, __type_call<_ApplyFn, _Vt>>;
+  using __apply_quals _CCCL_NODEBUG_ALIAS = __tuple_types<__type_call<_ApplyFn, __value_type<_Idx>>...>;
 };
 
 template <class _Tp,

--- a/libcudacxx/include/cuda/std/__tuple_dir/make_tuple_types.h
+++ b/libcudacxx/include/cuda/std/__tuple_dir/make_tuple_types.h
@@ -21,6 +21,7 @@
 #endif // no system header
 
 #include <cuda/std/__fwd/array.h>
+#include <cuda/std/__fwd/complex.h>
 #include <cuda/std/__fwd/tuple.h>
 #include <cuda/std/__tuple_dir/tuple_element.h>
 #include <cuda/std/__tuple_dir/tuple_indices.h>
@@ -61,7 +62,14 @@ struct __make_tuple_types_flat<array<_Vt, _Np>, __tuple_indices<_Idx...>>
   template <size_t>
   using __value_type = _Vt;
   template <class _Tp, class _ApplyFn = __apply_cvref_fn<_Tp>>
-  using __apply_quals = __tuple_types<__type_call<_ApplyFn, __value_type<_Idx>>...>;
+  using __apply_quals _CCCL_NODEBUG_ALIAS = __tuple_types<__type_call<_ApplyFn, __value_type<_Idx>>...>;
+};
+
+template <class _Vt>
+struct __make_tuple_types_flat<complex<_Vt>, __tuple_indices<0, 1>>
+{
+  template <class _Tp, class _ApplyFn = __apply_cvref_fn<_Tp>>
+  using __apply_quals _CCCL_NODEBUG_ALIAS = __tuple_types<__type_call<_ApplyFn, _Vt>, __type_call<_ApplyFn, _Vt>>;
 };
 
 template <class _Tp,

--- a/libcudacxx/include/cuda/std/__tuple_dir/tuple_like.h
+++ b/libcudacxx/include/cuda/std/__tuple_dir/tuple_like.h
@@ -20,6 +20,7 @@
 #  pragma system_header
 #endif // no system header
 
+#include <cuda/__fwd/complex.h>
 #include <cuda/std/__concepts/concept_macros.h>
 #include <cuda/std/__fwd/array.h>
 #include <cuda/std/__fwd/complex.h>
@@ -57,6 +58,9 @@ inline constexpr bool __tuple_like_impl<array<_Tp, _Size>> = true;
 
 template <class _Tp>
 inline constexpr bool __tuple_like_impl<complex<_Tp>> = true;
+
+template <class _Tp>
+inline constexpr bool __tuple_like_impl<::cuda::complex<_Tp>> = true;
 
 template <class _Ip, class _Sp, ::cuda::std::ranges::subrange_kind _Kp>
 inline constexpr bool __tuple_like_impl<::cuda::std::ranges::subrange<_Ip, _Sp, _Kp>> = true;

--- a/libcudacxx/include/cuda/std/__tuple_dir/tuple_like_ext.h
+++ b/libcudacxx/include/cuda/std/__tuple_dir/tuple_like_ext.h
@@ -21,6 +21,7 @@
 #endif // no system header
 
 #include <cuda/std/__fwd/array.h>
+#include <cuda/__fwd/complex.h>
 #include <cuda/std/__fwd/complex.h>
 #include <cuda/std/__fwd/pair.h>
 #include <cuda/std/__fwd/tuple.h>
@@ -53,6 +54,9 @@ inline constexpr bool __tuple_like_ext<array<_Tp, _Size>> = true;
 
 template <class _Tp>
 inline constexpr bool __tuple_like_ext<complex<_Tp>> = true;
+
+template <class _Tp>
+inline constexpr bool __tuple_like_ext<::cuda::complex<_Tp>> = true;
 
 template <class... _Tp>
 inline constexpr bool __tuple_like_ext<__tuple_types<_Tp...>> = true;

--- a/libcudacxx/include/cuda/std/__tuple_dir/tuple_like_ext.h
+++ b/libcudacxx/include/cuda/std/__tuple_dir/tuple_like_ext.h
@@ -20,8 +20,8 @@
 #  pragma system_header
 #endif // no system header
 
-#include <cuda/std/__fwd/array.h>
 #include <cuda/__fwd/complex.h>
+#include <cuda/std/__fwd/array.h>
 #include <cuda/std/__fwd/complex.h>
 #include <cuda/std/__fwd/pair.h>
 #include <cuda/std/__fwd/tuple.h>

